### DISCUSSION TITLE
Async fetch i18n config for each page request?

### DISCUSSION THREAD

**Discussion by sugoidesune** [nuxt-modules/i18n#1285] [Q&A]
I would like to load the config of i18n asynchronously for each page request.
As I use the same page for different domains it would need different configs.
Is this somehow possible?

```js
//nuxt.config.js

i18n: async () => await $api('i18nconfig')
/*{ 
    locales: ['en', 'fr', 'es'] ,
    defaultLocale: 'en',
    vueI18n: {
      fallbackLocale: 'en',
      messages: {
        en: {
          welcome: 'Welcome'
        },
        fr: {
          welcome: 'Bienvenue'
        },
        es: {
          welcome: 'Bienvenido'
        }
      }
    }
  }*/
}
```

**Comment by rchl** [comment]
Not possible.

Many of the options are evaluated at build time as they affect the extra routes that are generated so even if some options could be set dynamically like that, that would likely create issues and inconsistencies.
